replay: TestCompactionsQuiesce failure

This was under the regular go-linux build, and different error than in #2355.

https://github.com/cockroachdb/pebble/actions/runs/5881254195/job/15949354713?pr=2821

```
--- FAIL: TestCompactionsQuiesce (2.88s)
    replay_test.go:517: 
        	Error Trace:	replay_test.go:517
        	            				replay_test.go:524
        	Error:      	Received unexpected error:
        	            	sstable 000383 not found
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/pebble/replay.(*Runner).prepareWorkloadSteps.func4
        	            	  | 	/home/runner/work/pebble/pebble/replay/replay.go:831
        	            	  | github.com/cockroachdb/pebble/replay.(*Runner).prepareWorkloadSteps
        	            	  | 	/home/runner/work/pebble/pebble/replay/replay.go:884
        	            	  | github.com/cockroachdb/pebble/replay.(*Runner).Run.func1
        	            	  | 	/home/runner/work/pebble/pebble/replay/replay.go:371
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:[75](https://github.com/cockroachdb/pebble/actions/runs/5881254195/job/15949354713?pr=2821#step:4:76)
        	            	  | runtime.goexit
        	            	  | 	/opt/hostedtoolcache/go/1.20.7/x64/src/runtime/asm_amd64.s:1598
        	            	Wraps: (2) sstable 0003[83](https://github.com/cockroachdb/pebble/actions/runs/5881254195/job/15949354713?pr=2821#step:4:84) not found
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestCompactionsQuiesce
```

--- a/replay/replay_test.go
+++ b/replay/replay_test.go
@@ -571,7 +571,7 @@ func buildHeavyWorkload(t *testing.T) vfs.FS {
 		require.NoError(t, b.Commit(pebble.NoSync))
 		require.NoError(t, d.Flush())
 	}
-	wc.Stop()
+	wc.WaitAndStop()
 
 	defer d.Close()
 	return destFS

--- a/replay/workload_capture.go
+++ b/replay/workload_capture.go
@@ -383,6 +383,18 @@ func (w *WorkloadCollector) Start(destFS vfs.FS, destPath string) {
 	go w.copyFiles()
 }
 
+// WaitAndStop waits for all enqueued sstables to be copied over, and then
+// calls Stop. Gracefully ensures that all sstables referenced in the collected
+// manifest's latest version edit will exist in the copy directory.
+func (w *WorkloadCollector) WaitAndStop() {
+	w.mu.Lock()
+	for w.mu.tablesEnqueued != w.mu.tablesCopied {
+		w.mu.copyCond.Wait()
+	}
+	w.mu.Unlock()
+	w.Stop()
+}
+
 // Stop stops collection of the workload.
 func (w *WorkloadCollector) Stop() {
 	w.mu.Lock()

